mvcc/backend/batch_tx.go

```diff
@@ -323,6 +319,17 @@ func (t *batchTxBuffered) unsafeCommit(stop bool) {
 	}
 }
 
+func waitAndRollback(tx *bolt.Tx, wg *sync.WaitGroup, lg *zap.Logger) {
```

Comment:
`tx` and `wg` need to be passes into the function. `batchTxBuffered`'s `tx` and `wg` will be reset / re-created immediate after this function call, whereas this function waits until all ongoing reads using the input `tx` is done.

Reply:
Making this a receiver and moving the logger out of the params might make that even more clear.

Reply:
Did you mean something like `func (lg *zap.Logger) waitAndRollback(tx *bolt.Tx, wg *sync.WaitGroup)`? I have mixed feeling about this. By convention (in this repo), zap logger is passed to functions as input parameter.

Reply:
I was thinking more like `func (t *batchTxBuffered) waitAndRollback(tx *bolt.Tx, wg *sync.WaitGroup)` so that the only params that need to be passed are the ones that are semantically important to the call. But this is minor,

Reply:
Or just inline the function in `unsafeCommit`? it doesn't seem to benefit from being separate.

Reply:
Good idea.